File: Modules/Payments/Http/Resources/PaymentResource.php
<?php

namespace Modules\Payments\Http\Resources;

use Illuminate\Http\Resources\Json\JsonResource;
use Modules\Auth\Http\Resources\UserResource;
use Modules\Sessions\Http\Resources\SessionResource;

class PaymentResource extends JsonResource
{
    public function toArray($request): array
    {
        return [
            'id' => $this->id,
            'user_id' => $this->user_id,
            'session_id' => $this->session_id,
            'amount' => $this->amount,
            'currency' => $this->currency,
            'status' => $this->status,
            'gateway' => $this->gateway,
            'gateway_transaction_id' => $this->gateway_transaction_id,
            'gateway_response' => $this->gateway_response,
            'paid_at' => $this->paid_at,
            'created_at' => $this->created_at,
            'updated_at' => $this->updated_at,
            'user' => new UserResource($this->whenLoaded('user')),
            'session' => new SessionResource($this->whenLoaded('session')),
        ];
    }
}

File: Solid.Api/Features/Payments/PaymentResource.cs
using Solid.Api.Common;
using Solid.Api.Database.Entities;

namespace Solid.Api.Features.Payments;

public static class PaymentResource
{
    public static object From(Payment payment)
    {
        return new
        {
            id = payment.Id,
            user_id = payment.UserId,
            session_id = payment.SessionId,
            amount = payment.Amount,
            currency = payment.Currency,
            status = payment.Status,
            gateway = payment.Gateway,
            gateway_transaction_id = payment.GatewayTransactionId,
            gateway_response = JsonPayload.Parse(payment.GatewayResponse),
            paid_at = EgyptDateTime.Format(payment.PaidAt),
            created_at = EgyptDateTime.Format(payment.CreatedAt),
            updated_at = EgyptDateTime.Format(payment.UpdatedAt)
        };
    }
}
